llrt_modules/src/utils/array_buffer.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use rquickjs::{ArrayBuffer, Ctx, Error, FromJs, IntoJs, Object, Result, TypedArray, Value};
+
+#[cfg(feature = "buffer")]
+use crate::buffer::Buffer;
+
+pub struct ArrayBufferView<'js> {
+    value: Value<'js>,
+    buffer: ArrayBuffer<'js>,
+}
```

Comment:
So I did dig into the node code to see how all this was being implemented in v8. The name comes from a [v8 class actually] (https://v8.github.io/api/head/classv8_1_1ArrayBufferView.html). The idea is you need a shared abstraction for all objects that can hold an array buffer to be able to retrieve that buffer and still maintain the original value reference (since all read write methods need to return the input reference).

This is somewhat similar to the existing utils `get_bytes` in the cases it checks but it is more generic. Eventually we could use it to replace the method.

Reply:
That was my assumption. Very clever and simple implementation! What I think you might run into though is that sub arrays will share the underlying buffer. So you need to keep track of byte count and offsets on the typed array before returning the buffer to know where we're at, for example:
https://github.com/awslabs/llrt/blob/88355245ffd0bb8a5057636ea299ee052c91d81c/llrt_utils/src/bytes.rs#L113

Reply:
Yep, I made a note to fix that in Github. I realized that while reading the Buffer code!